.github/workflows/reusable-test.yaml

```diff
@@ -52,6 +55,15 @@ jobs:
         run: pip install poetry
 
       - uses: actions/cache@v3
+        if: inputs.os == 'ubuntu-latest' && inputs.python-version == '3.11'
+        with:
+          path: |
+            ~/.cache/pypoetry/virtualenvs
+            C:\Users\runneradmin\AppData\Local\pypoetry\Cache\virtualenvs
+          key: ${{ inputs.os }}-${{ steps.install-python.outputs.python-version }}-poetry-${{ hashFiles('poetry.lock') }}-${{ inputs.pandas-version }}
+
+      - uses: actions/cache@v3
+        if: (inputs.os == 'ubuntu-latest' && inputs.python-version == '3.9') || inputs.os == 'windows-latest'
```

Comment:
I'm not really happy about these if statements here. It'd be better if they weren't aware of which versions of things the external caller is using. 

Why is this necessary here? We should really have a comment  explaining that.